core/src/main/java/io/grpc/ServerMethodDefinition.java

```diff
@@ -31,67 +31,43 @@
 
 package io.grpc;
 
-import java.io.InputStream;
-
 /**
  * Definition of a method bound by a {@link io.grpc.HandlerRegistry} and exposed
  * by a {@link Server}.
  */
 public final class ServerMethodDefinition<RequestT, ResponseT> {
-  private final String name;
-  private final Marshaller<RequestT> requestMarshaller;
-  private final Marshaller<ResponseT> responseMarshaller;
+  private final MethodDescriptor<RequestT, ResponseT> method;
   private final ServerCallHandler<RequestT, ResponseT> handler;
 
   // ServerMethodDefinition has no form of public construction. It is only created within the
   // context of a ServerServiceDefinition.Builder.
-  ServerMethodDefinition(String name, Marshaller<RequestT> requestMarshaller,
-      Marshaller<ResponseT> responseMarshaller, ServerCallHandler<RequestT, ResponseT> handler) {
-    this.name = name;
-    this.requestMarshaller = requestMarshaller;
-    this.responseMarshaller = responseMarshaller;
+  ServerMethodDefinition(MethodDescriptor<RequestT, ResponseT> method,
+      ServerCallHandler<RequestT, ResponseT> handler) {
+    this.method = method;
     this.handler = handler;
   }
 
   /**
    * Create a new instance.
    *
-   * @param name the simple name of a method.
-   * @param requestMarshaller marshaller for request messages.
-   * @param responseMarshaller marshaller for response messages.
+   * @param method the {@link MethodDescriptor} for this method.
    * @param handler to dispatch calls to.
    * @return a new instance.
    */
   public static <RequestT, ResponseT> ServerMethodDefinition<RequestT, ResponseT> create(
-      String name, Marshaller<RequestT> requestMarshaller,
-      Marshaller<ResponseT> responseMarshaller, ServerCallHandler<RequestT, ResponseT> handler) {
-    return new ServerMethodDefinition<RequestT, ResponseT>(name, requestMarshaller,
-        responseMarshaller, handler);
+      MethodDescriptor<RequestT, ResponseT> method,
+      ServerCallHandler<RequestT, ResponseT> handler) {
+    return new ServerMethodDefinition<RequestT, ResponseT>(method, handler);
   }
 
-  /** The simple name of the method. It is not an absolute path. */
-  public String getName() {
-    return name;
+  /** The fully qualified name of the method. */
+  public String getFullName() {
```

Comment:
Should we get rid of this, too? It doesn't seem anyone is using it.  I guess it could be useful and is easy...